google/cloud/pubsub_v1/publisher/flow_controller.py

```diff
@@ -76,12 +76,12 @@ def add(self, message):
 
         Raises:
             :exception:`~pubsub_v1.publisher.exceptions.FlowControlLimitError`:
-                If adding a message exceeds flow control limits and the desired
-                action is :attr:`~google.cloud.pubsub_v1.types.LimitExceededBehavior.ERROR`.
-            :exception:`~pubsub_v1.publisher.exceptions.PermanentlyBlockedError`:
-                If adding a message exceeds total flow control limits and would
-                always overflow on its own, and the desired action is
-                :attr:`~google.cloud.pubsub_v1.types.LimitExceededBehavior.BLOCK`.
+                If adding a message would exceed flow control limits and the desired
+                action is :attr:`~google.cloud.pubsub_v1.types.LimitExceededBehavior.ERROR`,
+                or if a message would always exceed total flow control limits on
+                its own and the desired action is
```

Comment:
How about:

Raised when the desired action is :attr:`~google.cloud.pubsub_v1.types.LimitExceededBehavior.ERROR` and the message would exceed flow control limits, or when the desired action is :attr:`~google.cloud.pubsub_v1.types.LimitExceededBehavior.BLOCK` and the message would block forever against the flow control limits.

Reply:
I agree that it's hard to mention the size being the problem when there's a few other reasons as well.

Reply:
Changed, I also think the proposed version is worded better.